Source/DafnyCore/Rewriters/InductionHeuristic.cs

```diff
@@ -0,0 +1,175 @@
+using System.Diagnostics.Contracts;
```

Comment:
All code in this file was moved without semantic changes, from InductionRewriter.cs